bin/extract_codeobject

```diff
@@ -0,0 +1,426 @@
+#!/usr/bin/perl
+use strict;
+use File::Copy;
+use File::Spec;
+use File::Basename;
+use File::Which;
+use Cwd 'realpath';
+use Getopt::Std;
+use List::Util qw(max);
+
+sub usage {
+  print("Usage: $0 [OPTION]... -i <input>\n");
+  print("Extract the device kernels from an hcc executable.\n\n");
+  print("-h                                \tshow this help message\n");
+  print("-i <input>                        \t[DEPRECATED] input file\n");
+  print("-E <kernel-name>[:<kernel-name>]  \tExtract code object by URI name. Multiple code object names may be specified using \':\' delimiter.\n");
```

Comment:
kernel-name -> code-object-uri

I am not keen on the : sparator. Would it be better to just allow multiple -E arguments?

Would -u be a better option name as this is specifying a Uri?